src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,74 @@
-export const PeopleFilters = () => {
+import { Link, useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { getSearchWith } from '../utils/searchHelper';
+
+const centuriesArray = [
+  '16', '17', '18', '19', '20',
+];
+
+export const PeopleFilters: React.FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const centuries = searchParams.getAll('centuries') || [];
+  const sex = searchParams.get('sex') || 'All';
+  const query = searchParams.get('query') || '';
+
+  const toggleCenturies = (century: string) => {
+    if (!century) {
+      return getSearchWith(searchParams, {
+        centuries: [],
+      });
+    }
+
+    return getSearchWith(searchParams, {
+      centuries: centuries.includes(century)
+        ? centuries.filter(ch => ch !== century)
+        : [...centuries, century],
+    });
+  };
+
+  const toggleSex = (newSex: string | null) => {
+    if (newSex === 'All') {
+      return getSearchWith(searchParams, { sex: null });
+    }
+
+    return getSearchWith(searchParams, { sex: newSex });
+  };
+
+  const toggleQuery = (newQuery: string) => {
+    const search = getSearchWith(searchParams, { query: newQuery });
+
+    setSearchParams(search);
+  };
+
+  const toggleAllReset = () => {
+    return getSearchWith(
+      searchParams, { query: null, centuries: [], sex: null },
+    );
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+        <Link
+          className={cn({ 'is-active': sex === 'All' })}
+          to={{ search: toggleSex('All') }}
+        >
+          All
+        </Link>
+        <Link
+          className={cn({ 'is-active': sex === 'm' })}
```

Comment:
we can also use SearchLink

Reply:
For sure. Also render as a list (using `map`). Fix everywhere